azkaban-common/src/main/java/azkaban/executor/OnContainerizedExecutionEventListener.java

```diff
@@ -53,13 +56,41 @@ void restartExecutableFlow(final ExecutableFlow originalExFlow) {
       logger.error(e.getMessage());
       return;
     }
-    final ExecutableFlow retryExFlow =
+    ExecutableFlow retryExFlow =
         this.executorManagerAdapter.createExecutableFlow(project, flow);
+
+    final ExecutionOptions options = originalExFlow.getExecutionOptions();
+
+    final String restartStrategy = options.getFlowParameters().
+        getOrDefault(FLOW_PARAM_RESTART_STRATEGY, FlowRetryStrategy.DEFAULT.name()).trim();
+    logger.info(String.format("Retry execution of exec Id %d is set to use strategy [%s].",
+        originalExFlow.getExecutionId(), restartStrategy));
+    FlowRetryStrategy strategyEnum = FlowRetryStrategy.valueOf(restartStrategy);
+
+    // default strategy - not applying anything on the new one, so as like a new execution
+    if (restartStrategy.isEmpty() || strategyEnum.equals(FlowRetryStrategy.DEFAULT)) {
+      logger.info(String.format("Use default strategy when restarting the execution %s",
+          originalExFlow.getExecutionId()));
+    } else {
+      // non-default strategies
+      if (strategyEnum.equals(FlowRetryStrategy.DISABLE_SUCCEEDED_NODES)){
+        try {
+          disableSucceededSkippedJobsInRetryFlow(originalExFlow, retryExFlow);
```

Comment:
I think the `restartExecutableFlow` method handles both user and system retries. In the case of system retries  (for status PREPARING, DISPATCHING) the flow execution hasn't started yet so traversing the DAG to disable some nodes doesn't make much sense as non have yet run. As an optimization, is it possible to detect the type of retry and avoid the DAG traversing if it's a system retry? It can be addressed in a separate PR but can we add a TODO in the code perhaps?

Reply:
good call out, I will add this as TODO so as system-retry on `PREPARING, DISPATCHING` can be improved

Reply:
thanks!